alchemy/alchemy.py

```diff
@@ -1262,6 +1541,51 @@ def perturbSystem(cls, system, alchemical_state):
 
         return
 
+    def getEnergyComponents(self, alchemical_state, positions, box_vectors=None, use_all_parameters=True):
+        """
+        Compute potential energy by Force component for the corresponding alchemically-modified system.
+
+        Parameters
+        ----------
+        alchemical_state : AlchemicalState
+            The alchemical state to est the Context to.
+        use_all_parameters : bool, optional, default=False
+            If True, will ensure that all parameters are used or raise an Exception if not.
+
+        """
+        if not hasattr(self, 'alchemically_modified_system_with_force_groups'):
+            # Create deep copy of alchemical system.
+            import copy
+            system = copy.deepcopy(self.alchemically_modified_system)
+            # Separate all forces into separate force groups.
+            assert (system.getNumForces() < 16), "self.alchemically_modified_system has more than 16 force groups; can't compute individual force component energies."
```

Comment:
Taking a guess: 
16 forces, possibly all alchemically split between electrostatics and sterics, 32 possible forces (max number of `ForceGroup`s. However, we always need at least 1 group to drop all the non-alchemical forces, so we can only ever have 15 forces max.